### PR TITLE
Pract 57.UI kit

### PR DESCRIPTION
**Pull request by alejandroo-pc**
## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Acceptance Criteria

- [x] Accessible UI kit for consistent styling throughout the project.
- [x] Global access of css variables across the app 
- [x] Change the color of an element using an imported CSS variable.
- [x] Include a reusable `Button` component to maintain consistent styling across the project.

## Related Issue
[Jira ticket PRACT-57](https://practicum-team4.atlassian.net/jira/software/projects/PRACT/boards/2?selectedIssue=PRACT-57)

## Description

A unified UI kit has been implemented to ensure consistent styling across the application. These variables function as a centralized styling system, maintaining a cohesive design language throughout the entire application.The UI kit aligns with the wireframes outlined in [Figma.](https://www.figma.com/file/F7sNzfbrDwaWdslg5R5ZCS/PRACT-Flashcard-App?type=design&node-id=729-472&mode=design&t=ulN2itkU3FRWlNP3-0) 

A reusable button component that for consistent styling throughout the project. The button component can be easily imported and used across different components.

### How to Use Global-Styles
No need to import App.css into your file, it is applied globally. 
The css variables can be accessed directly into a .css file

For example, to set the color of a button:
```
.submit-button {
  background-color: var(--primary-button-background);
  border: solid var(--primary-button-border);
  color: var(--primary-button-text);
}
```

### How to use Button component
Import the Button component into the desired file:
`import Button from "./PATH/components/button/Button.js";`

Use the Button component within the JSX of your component:
```
<Button onClick={handleClick} type="submit">
  Click me
</Button>
```

### Before
Screenshot 2023-11-26 at 7 50 14 PM

### After
Screenshot 2023-11-26 at 6 51 09 PM

### CODE REVIEW

**Review by elenademi — APPROVED** [review]

File: src/pages/login/LoginPage.js [hunk 1]
```diff
@@ -60,10 +60,12 @@ function LoginPage() {
           type="password"
           name="password"
           id="password"
-        required
+          required
         ></input>
 
-        <button type="submit">Submit</button>
+        <Button className="submit-button" type="submit">
```
Comment (nikkigraybeal) [inline src/pages/login/LoginPage.js:66]: this `type="submit"` won't work how you have it. Right now it's being passed as a prop, which you are not receiving in the `<Button />` component so it's not actually being applied to the `<button>` element in the JSX of the Button component.

If you want to be able to determine the type of button you will need to pass type as a prop (which you are doing here!) AND make sure you are taking that prop in in the `<Button /> `component.

File: src/components/button/Button.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+// components/Button.js
+
+import React from "react";
+import "./Button.css";
+
+function Button({ children, className, onClick }) {
```
Comment (nikkigraybeal) [inline src/components/button/Button.js:6]: if you think we will be rendering components inside of our Buttons, then the children prop makes sense (maybe we will want to pass icons?) But if we are really just going to have simple buttons that just have text in them, I think we can skip receiving and rendering children in this component. 
you could just pass buttonText, className, clickHandler and type to the component as props. 
then use it like this:
`<Button buttonText="Submit" className="login-btn" clickHandler={handleLogin} type="submit" />`

**Review by nikkigraybeal — CHANGES_REQUESTED** [review]
Great work, Alejandro! This will make styling our app so much easier! Just a small thing to address with the reusable Button component and then I think it's good to merge!

File: src/pages/login/LoginPage.js (hunk 1 above)
Comment (Alejandro-Patino-Camargo) [reply src/pages/login/LoginPage.js:66]: Thats right I never added it as props. It's updated now!

**Review by nikkigraybeal — APPROVED** [review]
Awesome job, Alejandro! Looks good to me.

**Pull request merged**